# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/OffscreenCanvasDisplayHelper.cpp
# Commit: 5be8557c4721
# Full Hash: 5be8557c4721b4d1a2ac07bf5ae0516d9f2da2bc
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2021-12-10 09:10:34
# Regressor Bug: 1736177
# File Overlap Count: 2
# Description:
#   Bug 1736177 - Part 11. Plumb in OffscreenCanvasDisplayHelper. r=jgilbert
#   
#   This patch integrates OffscreenCanvasDisplayHelper with
#   HTMLCanvasElement, OffscreenCanvas and nsDisplayCanvas to allow
#   asynchronous display of an OffscreenCanvas.
# ==============================================================================

diff -r 49f8b4205a46 -r 5be8557c4721 dom/canvas/OffscreenCanvasDisplayHelper.cpp
--- a/dom/canvas/OffscreenCanvasDisplayHelper.cpp	Thu Dec 09 19:25:28 2021 +0000
+++ b/dom/canvas/OffscreenCanvasDisplayHelper.cpp	Thu Dec 09 19:25:28 2021 +0000
@@ -145,16 +145,20 @@
   MOZ_ASSERT(NS_IsMainThread());
 
   HTMLCanvasElement* canvasElement;
+  uint32_t width, height;
 
   {
     MutexAutoLock lock(mMutex);
     MOZ_ASSERT(mPendingInvalidate);
     mPendingInvalidate = false;
     canvasElement = mCanvasElement;
+    width = mWidth;
+    height = mHeight;
   }
 
   if (canvasElement) {
     SVGObserverUtils::InvalidateDirectRenderingObservers(canvasElement);
+    canvasElement->InvalidateCanvasPlaceholder(width, height);
     canvasElement->InvalidateCanvasContent(nullptr);
   }
 }